integration/rpc_test.go

```diff
@@ -229,38 +230,58 @@ func TestGetStateRPC(t *testing.T) {
 	ctx, ctxCancel := context.WithCancel(context.Background())
 	defer ctxCancel()
 
-	// retrieve the state and make sure everything looks correct
-	var grpcState *proto.State
+	// try a few times and wait around until we see the build is complete, or fail.
 	attempts = 0
+outerLoop:
```

Comment:
this is very complex and long, hard to understand, please refactor

Reply:
yeah sorry, dunno what i was smoking with this :laughing: should be more clear now